kubevela.yaml

```diff
@@ -16,6 +16,8 @@ environment:
       - busybox
       - ca-certificates-bundle
       - go
+  environment:
```

Comment:
we always want to link dynamically so we can omit CGO_ENABLED and while we're here we can also remove all the explicit build-time dependencies because go/build pipeline already pulls in all of those. 

ref: https://github.com/chainguard-dev/melange/blob/83d4d3ec5dc419946ab0510833d6e006b6b29797/pkg/build/pipelines/go/build.yaml#L3-L7